### DISCUSSION TITLE
Can i use this to encode multiple files into a single multilayer .avif?

### DISCUSSION THREAD

**Discussion by OWL4C** [Julian/avif#8] [General]
I use https://github.com/zzag/plasma5-wallpapers-dynamic for timed wallpapers on my KDE Plasma Desktop. The creator has chosen .avif as his file of choice, and provided a tool to build such an avif file from a metadata.json and the source pictures, however this tool is broken. I want to write my own python implementation of the builder tool, which is only a subset of the repo, it is more focused on the Desktop Integration/Plugin.

I cannot, however, find any docs on how to read, create or otherwise interact with the multilayer .avif files. Examples can be found at https://github.com/karmanyaahm/awesome-plasma5-dynamic-wallpapers
I have no experience with file formats and no idea on how to create an avif with multiple images, nor do i know how to reverse engineer these existing images to find out how they are structured. Any help would be appreciated!

**Comment by Julian** [comment]
I'm not sure! This library is a thin wrapper over libavif -- https://github.com/AOMediaCodec/libavif -- so I would try to see if you can find any information there, and then if so, that doesn't necessarily mean there's the thin wrapper needed to use it here in this repo, but if not, it's generally easy to add.

So yeah, sorry for no further detailed info but I'd start there!

**Reply by OWL4C** [reply]
Thank you! I was hoping you could tell me where to look for these functionalities within libavif, i struggle to find some sort of wiki / docs / manpage with the functions anywhere.

**Reply by OWL4C** [reply]
Ok by looking at the plasma repo i found https://github.com/zzag/plasma5-wallpapers-dynamic/blob/249a487e7dd55ca57f6831e60af082cc91f26ba6/src/lib/kdynamicwallpaperwriter.cpp#L16 which is what is creating the avif, and it is opening an avif encoder and then calling `avifEncoderAddImage` for every image, i think. That would be sensible enough, but i don't think your wrapper wraps the encoder, does it?

**Reply by OWL4C** [reply]
I think unless i am missing something i am better of just going with cpp and libavif directly.
